subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/RecompilationSpec.java

```diff
@@ -29,6 +29,7 @@ public class RecompilationSpec {
     private final Collection<String> classesToProcess = new LinkedHashSet<>();
     private final Collection<GeneratedResource> resourcesToGenerate = new LinkedHashSet<>();
     private String fullRebuildCause;
+    private boolean isIncrementalCompilationOfJavaModule;
```

Comment:
There is some data duplication indeed. I removed this from recompileSpec and simplified it a bit. I left it in `JavaCompileSpec` for now, since it's needed later when compiling. If we separate these two informations, we would need to set `isModule` also in cases when it's not incremental. So it needs some more work as you said.